drivers/gpio/gpio_rt1718s_alert.c

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT richtek_rt1718s_alert
+
+/**
+ * @file Driver for handling RS1718S chip alert.
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/gpio/rt1718s.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/util_macro.h>
+LOG_MODULE_DECLARE(gpio_rt1718s_alert, CONFIG_GPIO_LOG_LEVEL);
+
+/* Driver config */
+struct rt1718s_alert_config {
+	/* Alert GPIO pin */
+	const struct gpio_dt_spec irq_gpio;
+	/* RT1718S devices which share the same alert pin */
+	const struct device **rt1718s_dev;
+	/* Number of RT1718S devices on the alert pin */
+	uint32_t rt1718s_num;
+};
+
+/* Driver data */
+struct rt1718s_alert_data {
+	/* Alert handler device */
+	const struct device *alert_dev;
+	/* Alert pin callback */
+	struct gpio_callback gpio_cb;
+	/* Alert worker */
+	struct k_work alert_worker;
+};
+
+static void rt1718s_alert_callback(const struct device *dev, struct gpio_callback *cb,
+				   uint32_t pins)
+{
+	ARG_UNUSED(pins);
+	struct rt1718s_alert_data *data = CONTAINER_OF(cb, struct rt1718s_alert_data, gpio_cb);
+
+	k_work_submit(&data->alert_worker);
+}
+
+static void rt1718s_alert_worker(struct k_work *work)
+{
+	struct rt1718s_alert_data *const data =
+		CONTAINER_OF(work, struct rt1718s_alert_data, alert_worker);
+	const struct rt1718s_alert_config *const config = data->alert_dev->config;
+	struct rt1718s_data *data_rt1718s;
+	uint16_t alert, mask;
+
+	do {
+		/* RT1718S device handler */
+		for (int i = 0; i < config->rt1718s_num; i++) {
+			data_rt1718s = config->rt1718s_dev[i]->data;
```

Comment:
how about moving this block in its own function? Pass `config->rt1718s_dev[i]` as an argument, should save some indentation, make everything more compact.